sponge_cost.sage

```diff
@@ -0,0 +1,20 @@
+import math
+
+def grobner_complexity(state_size, alpha, rounds_full, rounds_partial):
+    num_vars = (state_size - 1) * rounds_full + rounds_partial
+    num_equations = num_vars
+    d_reg = (1 + num_equations * (alpha - 1)) // 2
+    return math.log(binomial(num_vars + d_reg, d_reg) ** 2, 2)
+
+security = 128
+
+def interpolation_rounds_lower_bound(state_size, alpha):
+    return 1 + security * math.log(2, alpha) + math.log(state_size, alpha)
+
+rounds_full = 8
+rounds_partial = 25
+state_size = 3 # the state size
+alpha = 17
+
+assert (rounds_full + rounds_partial >= interpolation_rounds_lower_bound(state_size, alpha))
```

Comment:
This doesn't account for padding, I am not sure if that is intentional?
If you wanted it to account for padding, the left hand side should be 
`(rounds_full / 1.25) + (rounds_partial / 1.075)`

Reply:
I get that with the paper's recommended security margin, you should have 30 partial rounds.

`(log(2, alpha) * lambda + log(t, 2)) - R_f = (log(2, 17) * 128 + log(3, 2)) - 6 = 26.9`
Rounding this to the next achievable number of partial rounds, we get 27 partial rounds without security margin.

Adding the security margin, I get `27 * 1.075 = 29.025` partial rounds, or rounding up to 30. (The 7.5% in the paper was arbitrary, so 7.4% / 29 rounds depends on your preference)